From 9c22190e61029a2ed69c8af13fea837b03589152 Mon Sep 17 00:00:00 2001
From: Matthias Reichl <hias@horus.com>
Date: Sat, 30 May 2015 16:23:57 +0200
Subject: [PATCH] cirrus hack: report that the cirrus card is a iec958 device

---
 xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp b/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
index 0120bd5..2b42a9a 100644
--- a/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
+++ b/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
@@ -1346,6 +1346,10 @@ void CAESinkALSA::EnumerateDevice(AEDeviceInfoList &list, const std::string &dev
     if (info.m_displayName == "snd_rpi_hifiberry_digi")
       info.m_deviceType = AE_DEVTYPE_IEC958;
 
+    // hack: also report cirrus card as an iec958 device
+    if (info.m_displayName == "snd_rpi_wsp")
+      info.m_deviceType = AE_DEVTYPE_IEC958;
+
     if (info.m_deviceType == AE_DEVTYPE_HDMI && info.m_displayName.size() > 5 &&
         info.m_displayName.substr(info.m_displayName.size()-5) == " HDMI")
     {
-- 
2.1.4

